addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/device/DSMRDevice.java

```diff
@@ -0,0 +1,481 @@
+                    handleShutdownDSMRDevice();
+                    break;
+                case STARTING:
+                    if (System.currentTimeMillis()
+                            - deviceStatus.stateEnteredTS > DSMRDeviceConstants.SERIAL_PORT_AUTO_DETECT_TIMEOUT) {
+                        /* Device did not receive telegrams for too long, switch port speed */
+                        handleDSMRDeviceEvent(DSMRDeviceEvent.SWITCH_BAUDRATE,
+                                DSMRDeviceEvent.SWITCH_BAUDRATE.eventDetails);
+                    }
+                    break;
+                case SWITCH_PORT_SPEED:
+                    // Close current port
+                    dsmrPort.close();
+
+                    // Switch port settings first (otherwise initialize could use current port speed settings)
+                    dsmrPort.switchPortSpeed();
+
+                    // Open port again
+                    dsmrPort.open();
+                    break;
+                default:
+                    logger.warn("Unknown state {}", deviceStatus.deviceState);
+
+                    break;
+            }
+            if (deviceStatus.receivedCosemObjects.size() > 0) {
+                logger.debug("Dropping {} Cosem objects due to state {}", deviceStatus.receivedCosemObjects.size(),
+                        deviceStatus.deviceState);
+            }
+            deviceStatus.receivedCosemObjects.clear();
+        }
+    }
+
+    /**
+     * Send cosemObjects to the available meters
+     * This method will iterate through available meters and notify them about
+     * the CosemObjects received.
+     *
+     * @param cosemObjects. The list of {@link CosemObjects} received.
+     */
+    private void sendCosemObjects(List<CosemObject> cosemObjects) {
+        // Iterate through available meters and let them process the cosemObjects
+        for (DSMRMeter meter : availableMeters) {
+            logger.debug("Processing CosemObjects for meter {}", meter);
+            List<CosemObject> processedCosemObjects = meter.handleCosemObjects(cosemObjects);
+            logger.debug("Processed cosemObjects {}", processedCosemObjects);
+            cosemObjects.removeAll(processedCosemObjects);
+        }
+
+        /*
+         * There are still unhandled cosemObjects. Start discovery of DSMR meters
+         */
+        if (cosemObjects.size() > 0) {
+            logger.info("There are unhandled CosemObjects, start autodetecting meters");
+
+            DSMRMeterDetector meterDetector = new DSMRMeterDetector();
+            List<DSMRMeterDescriptor> detectedMeters = meterDetector.detectMeters(cosemObjects);
+            logger.info("Detected the following new meters: {}", detectedMeters.toString());
+
+            if (discoveryListener != null) {
+                for (DSMRMeterDescriptor meterDescriptor : detectedMeters) {
+                    if (discoveryListener.meterDiscovered(meterDescriptor)) {
+                    } else {
```

Comment:
Empty if body, why, please add a comment in the code to explain, maybe make it more explicit:

```java
boolean success = discoveryListener.meterDiscovered(meterDescriptor);
if (!success) {
    logger.info("DiscoveryListener {} rejected meter descriptor {}", discoveryListener, meterDescriptor);
}
```